frame/frame-layer.go

```diff
@@ -60,6 +60,19 @@ func NewFrameLayer(transportLayer io.ReadWriter) *Layer {
 	go frameLayer.bgWork()
 	go frameLayer.bgRead()
 
+	// Empty the buffer
+empty:
+	for {
+		select {
+		case <-parserInput:
```

Comment:
It looks like the reader may be backed by `bufio.Reader`. Could we expand the interface to use the Reset method?